src/bridge.ts

```diff
@@ -12,13 +12,14 @@ import { getBinaryExtension } from './platform.js'
 const DENO_VERSION_FILE = 'version.txt'
 const DENO_VERSION_RANGE = '^1.20.3'
 
-type LifecycleHook = () => void | Promise<void>
+type OnBeforeDownloadHook = () => void | Promise<void>
+type OnAfterDownloadHook = (hasError: boolean) => void | Promise<void>
```

Comment:
If we're telling the callback that there is an error, I think we might as well send it.

```suggestion
type OnAfterDownloadHook = (error: error | null) => void | Promise<void>
```